### PR TITLE
guix: Simplify code for Linux builds

### PR DESCRIPTION
**Pull request by hebasto**
While working on bitcoin/bitcoin#22458 I found that code for Linux builds could be simplified.

#### Guix builds on `x86_64`:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
b8b5d3c516e44f8248a03cee98bf556bd9ec36f8262e15b380a958eca19254c2  guix-build-d9da39fa37ee/output/aarch64-linux-gnu/SHA256SUMS.part
1db5939444af3f9c1f863ac81710b8062535fdb4fc41523ee61f40ba7a571d9c  guix-build-d9da39fa37ee/output/aarch64-linux-gnu/bitcoin-d9da39fa37ee-aarch64-linux-gnu-debug.tar.gz
ceec312f80bb6f42d8bfb28b373661fb7912a7b6ef4d10c151bdbac5d89385e3  guix-build-d9da39fa37ee/output/aarch64-linux-gnu/bitcoin-d9da39fa37ee-aarch64-linux-gnu.tar.gz
c232d9933c5062b0d6d7fa71761cdf872f6c49de0426e229b74a44d61bf9d3dd  guix-build-d9da39fa37ee/output/arm-linux-gnueabihf/SHA256SUMS.part
60c36fcd6fa5e58c243cf1c9dc3be383b5a9ec2bbfe53110133776ee79a7b446  guix-build-d9da39fa37ee/output/arm-linux-gnueabihf/bitcoin-d9da39fa37ee-arm-linux-gnueabihf-debug.tar.gz
e149bd724d9cf18dd0f38d2326c6bee90160cc4006a1ca806834f919029f58e1  guix-build-d9da39fa37ee/output/arm-linux-gnueabihf/bitcoin-d9da39fa37ee-arm-linux-gnueabihf.tar.gz
95e22b4ede04b61f43f54dc2116d1f4ac12534d987148d27bec6d6a631ea889e  guix-build-d9da39fa37ee/output/dist-archive/bitcoin-d9da39fa37ee.tar.gz
ff22ec01ffd21506a8e089664ff55cb76d9fd96301062056c0dcd35c35a428f5  guix-build-d9da39fa37ee/output/powerpc64-linux-gnu/SHA256SUMS.part
462f1f6ed2991cdcf37e8986ca82402506c3daab6af099258a66fc8b345cb0ea  guix-build-d9da39fa37ee/output/powerpc64-linux-gnu/bitcoin-d9da39fa37ee-powerpc64-linux-gnu-debug.tar.gz
7237c94ce85478830e847f12ba8cce575a98abb5e5a2d9e8a389983e2e905dad  guix-build-d9da39fa37ee/output/powerpc64-linux-gnu/bitcoin-d9da39fa37ee-powerpc64-linux-gnu.tar.gz
b52709a327ed5347af61cf8125a0bf3d86b02cb15f653829475be69781308e28  guix-build-d9da39fa37ee/output/powerpc64le-linux-gnu/SHA256SUMS.part
20f60ab1024c160858095cc7f8e0b140dbbfdd1fe365caeeaa255ac40a768452  guix-build-d9da39fa37ee/output/powerpc64le-linux-gnu/bitcoin-d9da39fa37ee-powerpc64le-linux-gnu-debug.tar.gz
a25b7cffe46f7adaba97e3ca891dc0226e47234a02f043929f462ce59cb3f368  guix-build-d9da39fa37ee/output/powerpc64le-linux-gnu/bitcoin-d9da39fa37ee-powerpc64le-linux-gnu.tar.gz
8698461bca56e0420abd78b4d1f32673cebafff4716c93f7b66426937ac384be  guix-build-d9da39fa37ee/output/riscv64-linux-gnu/SHA256SUMS.part
7a2d41c257ef11d859ec9b3bd3166a46ac36c920c12c5f92d6ec57ee0bf1cf90  guix-build-d9da39fa37ee/output/riscv64-linux-gnu/bitcoin-d9da39fa37ee-riscv64-linux-gnu-debug.tar.gz
c95c23e3ba3d8f0361e107cfa3118f4f124c9d89bdf8f1eca5141bc352b9c0af  guix-build-d9da39fa37ee/output/riscv64-linux-gnu/bitcoin-d9da39fa37ee-riscv64-linux-gnu.tar.gz
cebd2596daf488632aee3e5761afbb06eee6f331648719c38251d361261582e3  guix-build-d9da39fa37ee/output/x86_64-linux-gnu/SHA256SUMS.part
63796ad8d6621acc767bf21abcf8668d1e351ee6e95c932a48bb42fa01022e75  guix-build-d9da39fa37ee/output/x86_64-linux-gnu/bitcoin-d9da39fa37ee-x86_64-linux-gnu-debug.tar.gz
645500a034378b6ae64db062acbd1cc7cf61cbf549d09e6b94dcde44584e819b  guix-build-d9da39fa37ee/output/x86_64-linux-gnu/bitcoin-d9da39fa37ee-x86_64-linux-gnu.tar.gz
```

### CODE REVIEW

**Comment by fanquake** [comment]
d9da39fa37ee9db5972c20c6267d0386953fde9a looks fine. Can you explain the other changes? When did this code become redundant?

**Comment by hebasto** [comment]
> Can you explain the other changes?

Isn't a successful build an explanation?

> When did this code become redundant?

Not sure about that. Didn't check these changes with previous code.

**Comment by laanwj** [comment]
Concept ACK, looks like a good cleanup to me.

> Isn't a successful build an explanation?

No I don't think that's an explanation. I mean, worst case it breaks the generated binaries in some subtle way. Not saying it does, of course.

File: contrib/guix/libexec/build.sh [hunk 1]
```diff
@@ -104,16 +104,12 @@ case "$HOST" in
         ;;
     *linux*)
         CROSS_GLIBC="$(store_path "glibc-cross-${HOST}")"
-        CROSS_GLIBC_STATIC="$(store_path "glibc-cross-${HOST}" static)"
```
Comment (laanwj) [inline contrib/guix/libexec/build.sh:107]: I wonder why was this here, I don't think we ever needed a static glibc?

**Comment by jarolrod** [comment]
GUIX hashes, arm64:

```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

c55a1bd6ccf988eaee0af65ae14373be146e19a422e2896362419125b0d8675f  guix-build-d9da39fa37ee/output/arm-linux-gnueabihf/SHA256SUMS.part
6438a522f00d212fe9782035162d1309d3ab1d70d093a80d01a9d4f6bf4e6cf4  guix-build-d9da39fa37ee/output/arm-linux-gnueabihf/bitcoin-d9da39fa37ee-arm-linux-gnueabihf-debug.tar.gz
a52bbbed18d647611d3bc3bdee1ed05fa6301d28cf41b7934e209a738e26ea11  guix-build-d9da39fa37ee/output/arm-linux-gnueabihf/bitcoin-d9da39fa37ee-arm-linux-gnueabihf.tar.gz
95e22b4ede04b61f43f54dc2116d1f4ac12534d987148d27bec6d6a631ea889e  guix-build-d9da39fa37ee/output/dist-archive/bitcoin-d9da39fa37ee.tar.gz
42f5c15005fb0c9e1532fad1962ce6a6b9fc09797eecb12738cbb78b21b5ccad  guix-build-d9da39fa37ee/output/powerpc64-linux-gnu/SHA256SUMS.part
8c6d010b0ef261057c12d2c6d4b7e45ac62bfc02fca5b39845dfbf0afa91c22d  guix-build-d9da39fa37ee/output/powerpc64-linux-gnu/bitcoin-d9da39fa37ee-powerpc64-linux-gnu-debug.tar.gz
053c03090a36a4840bc528e75976ffda30ddf2f045721273eeb7c3817a948c77  guix-build-d9da39fa37ee/output/powerpc64-linux-gnu/bitcoin-d9da39fa37ee-powerpc64-linux-gnu.tar.gz
a2d6e9b9267333f6a3d87c8424b6f01885b561152426ecf9a2ecbced5bc8185e  guix-build-d9da39fa37ee/output/powerpc64le-linux-gnu/SHA256SUMS.part
adc0c5bd3e1e5ffa51dd2009b3e8134b996d22978bd16ecaccbf51660303ff66  guix-build-d9da39fa37ee/output/powerpc64le-linux-gnu/bitcoin-d9da39fa37ee-powerpc64le-linux-gnu-debug.tar.gz
165bcdc468c235efef1eeb24b7381abe86962c49407dd8f0be5caa81a78b576c  guix-build-d9da39fa37ee/output/powerpc64le-linux-gnu/bitcoin-d9da39fa37ee-powerpc64le-linux-gnu.tar.gz
0c8bdc5d0f8b33b602c3348447b6d4326767bddd123fbc6ce9ab3cf68094cfeb  guix-build-d9da39fa37ee/output/riscv64-linux-gnu/SHA256SUMS.part
2f1c57cf1668ca8b5b58e3423aba396889585e38fd3d34c60f75675260eb3ffa  guix-build-d9da39fa37ee/output/riscv64-linux-gnu/bitcoin-d9da39fa37ee-riscv64-linux-gnu-debug.tar.gz
2c69f17b3be8fdcdbc67d70bd89663083ce372a7e130fbb3d7646b155e4ff641  guix-build-d9da39fa37ee/output/riscv64-linux-gnu/bitcoin-d9da39fa37ee-riscv64-linux-gnu.tar.gz
d6ef9e6571c030d5d9bc720eeec0b190b4900a16fdc446288efbbe6d65161071  guix-build-d9da39fa37ee/output/x86_64-linux-gnu/SHA256SUMS.part
0bcdf19e4f1c7bfcddfbb3ff0f1622004350e113f56c851a649ccbaef712ec03  guix-build-d9da39fa37ee/output/x86_64-linux-gnu/bitcoin-d9da39fa37ee-x86_64-linux-gnu-debug.tar.gz
8234ead057a1e2ffb77b516bbbc07671dde4e39aad327aa956dc155c170dd8e2  guix-build-d9da39fa37ee/output/x86_64-linux-gnu/bitcoin-d9da39fa37ee-x86_64-linux-gnu.tar.gz
```
